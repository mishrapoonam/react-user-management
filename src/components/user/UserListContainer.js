import React, {Component} from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as userAction from '../../action/UserAction';
import UserList from './UserList';



export class UserListContainer extends Component {

    constructor() {
        super();
        this.handleDelete = this.handleDelete.bind(this);
    }


    componentDidMount() {
        this.props.action.getUsersAction()
            .catch(error => {
                console.log(error);
            });
    }



    handleDelete(id) {
        if (id!=null) {                     
            this.props.action.deleteUserAction(id)
                .catch(error => {
                    console.log(error);
                });
        }
    }


    render() {
        const { users } = this.props;
        return (
            <div className="container-fluid">
                <div className="row mt-3">
                    <div className="col">
                        <h3>Users List</h3>                        
                    </div>
                </div>

                <div className="row">
                    <div className="col">
                        <UserList users={users} handleDelete={this.handleDelete}/>
                    </div>
                </div>
            </div>
        );
    }
}



const mapStateToProps = state => ({
    users: state.userReducer.users
});



const mapDispatchToProps = dispatch => ({
    action: bindActionCreators(userAction, dispatch)

});



UserListContainer.propTypes = {
    users: PropTypes.array,
    action: PropTypes.object.isRequired,
};



export default connect(mapStateToProps, mapDispatchToProps)(UserListContainer);
