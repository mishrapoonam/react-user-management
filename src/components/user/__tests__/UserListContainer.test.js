import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { UserListContainer } from '../UserListContainer';


describe('UserListContainer.test.js', () => {

    it('renders without crashing', () => {
        const props = {
            users: [
                { title: 'Testing User array data' }
            ],
            action: { getUsersAction: jest.fn() }
        };

        const wrapper = shallow(<UserListContainer {...props} />);

        expect(wrapper).toHaveLength(1);

        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });

  
});

