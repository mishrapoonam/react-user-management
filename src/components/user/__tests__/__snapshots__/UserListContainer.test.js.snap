// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserListContainer.test.js renders without crashing 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row mt-3"
  >
    <div
      className="col"
    >
      <h3>
        Users List
      </h3>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <UserList
        handleDelete={[Function]}
        users={
          Array [
            Object {
              "title": "Testing User array data",
            },
          ]
        }
      />
    </div>
  </div>
</div>
`;
